src/main/scala/com/rbmhtechnology/calliope/EventReader.scala

```diff
@@ -29,7 +30,7 @@ trait EventReader[A] {
   * Mixin for applying an arbitrary gap-detection to an [[EventReader]].
   */
 trait GapDetectionEventReader[A] extends EventReader[A] with GapDetection {
```

Comment:
Although not part of this pull request, the type hierarchy around `GapDetection` feels a bit complicated. There is only a single algorithm at the moment and this could be an implementation detail of a concrete `EventReader` implementation. Also, the type name `ReaderTimestampGapDetection` is misleading IMO: it implies an algorithm (which is correct) but also inherits the `EventReader` interface (which is confusing). Alternatively, consider usage of type classes for detecting gaps on sequences.